python_anvil/api_resources/requests.py

```diff
@@ -19,14 +34,46 @@ def handle_error(self, response, status_code, headers):
 
         raise Exception(message)
 
-    def process_response(self, response, status_code, headers):
+    def process_response(self, response, status_code, headers, **kwargs):
         res = response
         if not 200 <= status_code < 300:
             self.handle_error(response, status_code, headers)
 
+        debug = kwargs.pop('debug', False)
+
+        # Include headers alongside the response.
+        # This is useful for figuring out rate limits outside of this library's
+        # scope and to manage waiting.
+        include_headers = kwargs.pop('include_headers', False)
+        if debug or include_headers:
+            return res, headers
```

Comment:
Any reason not to make this return an object/dict starting now? Certainly seems to make sense in the long-term, and if the user would have to explicitly alter the return behavior by passing `include_headers=True` they may as well deal with a `dict` instead of a `tuple`, no? This way the code is backwards compatible, and you can avoid a future breaking change since returning a `tuple` is sort of an odd pattern to continue/expand...

Reply:
I think you're right. Was trying to justify the tuple returns (I've seen a number of libraries do the same), but it does make sense to do it now.